balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -548,7 +575,12 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 		return
 	}
 
+	if oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.TransientFailure {
+		pickFirstConnectionAttemptsFailed.Record(b.metricsRecorder, 1, b.target)
+	}
```

Comment:
I don't understand your suggestion of moving this to a "select arms". I see two connectivity.TransientFailure conditions: one if it's in first pass, and one if it's in past the first pass.

Oh, so you want me to emit this from those two cases rather than if tf { record metric } at this point.

Reply:
Also, as I wrote to Arjan above, is there a harm in keeping the oldState Connecting check, even if there as an invariant that TF can only come after connecting? That makes it clearer to me.

Reply:
Removed old state check.